state/state.go

```diff
@@ -772,12 +805,14 @@ func (s *State) fetchMember(gID discord.GuildID, uID discord.UserID) (m *discord
 // tracksMessage reports whether the state would track the passed message and
 // messages from the same channel.
 func (s *State) tracksMessage(m *discord.Message) bool {
-	return (m.GuildID.IsValid() && s.Gateway.HasIntents(gateway.IntentGuildMessages)) ||
-		(!m.GuildID.IsValid() && s.Gateway.HasIntents(gateway.IntentDirectMessages))
+	return (m.GuildID.IsValid() &&
+		s.ShardManager.FromGuildID(m.GuildID).HasIntents(gateway.IntentGuildMessages)) ||
+		(!m.GuildID.IsValid() &&
+			s.ShardManager.FromGuildID(m.GuildID).HasIntents(gateway.IntentDirectMessages))
```

Comment:
Might be worth lifting `s.ShardManager.FromGuildID(m.GuildID)` into a variable to maintain code similar to the old code.